### PR TITLE
Add data_set_external_id parameter

### PR DESCRIPTION
**Pull request by haakonvt**

### CODE REVIEW

**Review by pavelzubarev — APPROVED** [review]
LGTM

File: src/function.py [hunk 1]
```diff
@@ -109,13 +123,19 @@ def zip_and_upload_folder(client: CogniteClient, config: FunctionConfig, name: s
                 logger.info(f"Added common directory: '{config.common_folder_path}' to the function")
                 _write_files_to_zip_buffer(zf, directory=config.common_folder_path)
 
-    file_meta = client.files.upload_bytes(buf.getvalue(), name=name, external_id=name)
+    data_set_id = None
+    if config.data_set_external_id is not None:
+        # This looks idiotic, but the SDK does not yet support ext_ids for files... facepalm:
+        data_set_id = get_data_set_id_from_external_id(client, config.data_set_external_id)
+
+    file_meta = client.files.upload_bytes(buf.getvalue(), name=name, external_id=name, data_set_id=data_set_id)
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply src/function.py:129]: Is the error one of the ones we retry?

[the remark this answers is not in the document]
Comment (pavelzubarev) [reply src/function.py:129]: SDK won't retry those, but we could retry that outside and that will be seen in the end as an Unauthorized error

[the remark this answers is not in the document]
Comment (haakonvt) [reply src/function.py:132]: ...but ´@retry´ will not retry them?

File: src/function.py [hunk 2]
```diff
@@ -25,6 +26,19 @@ class FunctionDeployError(Exception):
...
+def get_data_set_id_from_external_id(client: CogniteClient, ext_id: str) -> int:
+    """
+    Assuming internal IDs eventually will (read: should) die, we enforce the use
+    of external IDs in this Github action... but since the SDK (cur 2.10.3)
+    does not support data set external ID for FilesAPI, we need lookup...
+    """
+    if (ds := client.data_sets.retrieve(external_id=ext_id)):
+        return ds.id
+    raise CogniteNotFoundError(
+        f"Data set external ID: '{ext_id}' not found -OR- you don't have access to it!"
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply src/function.py]: Nice, added it - except fixed spelling ;P

File: src/function.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pavelzubarev) [reply src/function.py:132]: No. it should not

[the remark this answers is not in the document]
Comment (haakonvt) [reply src/function.py:132]: Nice, then I'm happy to just merge and not catch it. Ok with you?

[the remark this answers is not in the document]
Comment (pavelzubarev) [reply src/function.py:132]: Yes

File: src/function.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pavelzubarev) [reply src/function.py]: :pavel-rolling-eyes:

**Pull request merged**